js/index.js

```diff
@@ -1,42 +1,73 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+  let price = product.querySelector('.price span').innerText;
+  let quantity = product.querySelector('.quantity input').value;
+  let subTotal = price * quantity;
+
+  product.querySelector('.subtotal span').innerText = subTotal;
+
+  return subTotal;
 
   //... your code goes here
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
+  const singleProduct = document.querySelectorAll('.product');
+  const totalValueElement = document.querySelector('#total-value span');
+  let total = 0;
+  for (let i = 0; i < singleProduct.length; i++) {
+    total += updateSubtotal(singleProduct[i]);
+  }
 
-  // ITERATION 3
-  //... your code goes here
+  totalValueElement.innerText = total;
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
-  const target = event.currentTarget;
-  console.log('The target in remove is:', target);
+  const target = event.currentTarget.parentNode.parentNode;
+  target.remove();
+  calculateAll();
   //... your code goes here
 }
 
 // ITERATION 5
 
-function createProduct() {
-  //... your code goes here
+function createProduct(name) {
+  const cart = document.querySelector(`#cart tbody`);
+  const product = `<tr class="product">
+  <td class="name">
+    <span>${
+      name.currentTarget.parentNode.parentNode.querySelector('#name').value
+    }</span>
+  </td>
+  <td class="price">$<span>${
+    name.currentTarget.parentNode.parentNode.querySelector('#price').value
+  }</span></td>
+  <td class="quantity">
+    <input type="number" value="0" min="0" placeholder="Quantity" />
+  </td>
+  <td class="subtotal">$<span>0</span></td>
+  <td class="action">
+    <button class="btn btn-remove" onclick='removeProduct(event)'>Remove</button>
+  </td>
+</tr>`;
+  cart.insertAdjacentHTML('beforeend', product);
 }
```

Comment:
Aqui (ainda na função create) só faltou incluir o eventListner pros novos produtos criados (make sure remove button inherits the same behavior as other remove buttons) e limpar os campos depois de adicionado o novo produto (clean the fields)

Reply:
Eu passei a função inline na linha 55, acabei usando o onclick ao invés do  addEventListner